libraries/net/eth/EthernetInterface/EthernetInterface.h

```diff
@@ -45,7 +45,7 @@ class EthernetInterface {
   * \param gateway the gateway to use
   * \return 0 on success, a negative number on failure
   */
-  static int init(const char* ip, const char* mask, const char* gateway);
+  static int init(const char* ip, const char* mask, const char* gateway, const char * dns = NULL);
```

Comment:
This API change appears orthogonal to the issue being discussed - maybe it should be a separate PR?. However, it seems like a good idea to me.